pkg/service/providerindex/legacy.go

```diff
@@ -55,16 +59,29 @@ func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStor
 	}
 
 	return LegacyClaimsStore{
-		contentToClaims: contentToClaimsMapper,
+		contentToClaims: contentToClaimsMappers,
 		claimsStore:     claimStore,
 		claimsAddr:      claimsAddr,
 	}, nil
 }
 
 // Find looks for the corresponding claims for a given content hash in the mapper and then fetches the claims from the
 // claims store
-func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Multihash) ([]model.ProviderResult, error) {
-	claimsCids, err := ls.contentToClaims.GetClaims(ctx, contentHash)
+func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Multihash, targetClaims []multicodec.Code) ([]model.ProviderResult, error) {
+	for _, mapper := range ls.contentToClaims {
+		claims, err := ls.findOne(ctx, contentHash, targetClaims, mapper)
+		if err != nil {
+			return nil, err
+		}
+		if len(claims) > 0 {
+			return claims, nil
+		}
+	}
+	return nil, nil
```

Comment:
np: while equivalent in this case, returning an empty array feels more idiomatic IMO
```suggestion
	return []model.ProviderResult{}, nil
```